src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -181,6 +181,27 @@ public virtual void InitializeMining()
             startAfter: TimeSpans.TenSeconds);
         }
 
+        private IFederationMember DetermineExpectedMinerForTimestamp(uint headerTime)
+        {
+            // When blocks are missing for given timestamps we have to determine whom was supposed to mine
+            // by looking at the federation make-up at the time and whom mined last.
+
+            ChainedHeader prevBlockMined = this.consensusManager.Tip;
+            while (prevBlockMined.Header.Time >= headerTime)
```

Comment:
Shouldn't it just be greater than here (as opposed to greater than or equals to)? Cause if the current slot you want to check matches the chain tip time then we still set it back to previous?

Reply:
Done